typescript/lib/config.ts

```diff
@@ -117,6 +117,25 @@ export class AIConfigRuntime implements AIConfig {
     );
 
     Object.assign(aiConfig, remainingProps);
+
+    // Update the ModelParserRegistry with any model parsers specified in the AIConfig
+    if (aiConfig.metadata.model_parsers) {
+      for (const [modelName, modelParserId] of Object.entries(
+        aiConfig.metadata.model_parsers
+      )) {
+        const modelParser = ModelParserRegistry.getModelParser(modelParserId);
+        if (!modelParser) {
+          throw new Error(
+            `E1001: Unable to load AIConfig: It specifies ${JSON.stringify(
+              aiConfig.metadata.model_parsers
+            )}, but ModelParser ${modelParserId} for model ${modelName} does not exist. Make sure you have registered the ModelParser before loading the AIConfig.`
+          );
+        }
+
+        ModelParserRegistry.registerModelParser(modelParser, [modelName]);
```

Comment:
Should we consider updating the `modelParser` ID to `modelName` before registering it? ie `modelParser.id=modelName`

This might not be necessary if ModelParser never needs an explicit reference to ModelName since it usually has a reference to the prompt data.

Reply:
Resolved offline -- basically now we have a clearer definition of model parser ID vs. model ID. Both are needed.